omnigibson/object_states/object_state_base.py

```diff
@@ -67,23 +67,37 @@ def reset(self):
         """
         Resets this object state. By default, it clears all internal caching data
         """
-        self._cache = OrderedDict()
-        self._changed = OrderedDict()
+        self._cache = dict()
+        self._changed = dict()
+        self._last_t_updated = -1
 
     @property
     def cache(self):
         """
         Returns:
-            OrdereDict: Dictionary mapping specific argument combinations from @self.get_value() to cached values and
+            dict: Dictionary mapping specific argument combinations from @self.get_value() to cached values and
                 information stored for that specific combination
         """
         return self._cache
 
+    @classproperty
+    def requires_update(cls):
+        """
+        Returns:
+            bool: True if this state requires an update or not every sim step
+        """
+        # The __repr__ call tells us which actual function is being called
+        # Any subclass that actually implements _update() will cause this to return its own class_name._update
+        # Otherwise, __repr__ will return BaseObjectState._update
+        # So we use this as a heuristic to check whether or not a subclass has overridden the _update method
+        return "BaseObjectState" not in cls._update.__repr__()
```

Comment:
hmm this seems kind of hacky... can we find an alternative?

Reply:
UpdatingState?

Reply:
done!